server/lib/chat/chat-service.ts

```diff
@@ -490,41 +519,48 @@ export default class ChatService {
     })
   }
 
-  async getChannelInfo(channelId: SbChannelId, userId: SbUserId): Promise<ChannelInfo> {
+  async getChannelInfo(channelId: SbChannelId, userId: SbUserId): Promise<GetChannelInfoResponse> {
     const channelInfo = await getChannelInfo(channelId)
+    // TODO(2Pac): Get this from DB instead.
+    const isUserInChannel = this.state.users.get(userId)?.has(channelId)
 
     if (!channelInfo) {
       throw new ChatServiceError(ChatServiceErrorCode.ChannelNotFound, 'Channel not found')
     }
 
-    let userCount
-    if (!channelInfo.private || this.state.users.get(userId)?.has(channelId)) {
-      userCount = channelInfo.userCount
-    }
-
     return {
-      id: channelInfo.id,
-      name: channelInfo.name,
-      private: channelInfo.private,
-      official: channelInfo.official,
-      userCount,
+      basicChannelInfo: {
+        id: channelInfo.id,
+        name: channelInfo.name,
+      },
+      detailedChannelInfo: {
```

Comment:
As stated in the common file, this should either be all present or not present at all. Should probably also write a TODO here for the issue you filed about actually filling out the other fields (because this code does not make it clear that it has skipped a bunch of optional ones)